modules/launcher/src/test/java/org/mule/module/launcher/DeploymentServiceTestCase.java

```diff
@@ -2215,6 +2215,25 @@ public void redeploysExplodedDomainAfterDeploymentError() throws Exception
         assertEquals("Failed domain still appears as zombie after a successful redeploy", 0, deploymentService.getZombieDomains().size());
     }
 
+
+    @Test
+    public void deployFailsWhenMissingFile() throws Exception
+    {
+        addExplodedAppFromBuilder(emptyAppFileBuilder);
+
+        deploymentService.start();
+
+        assertApplicationDeploymentSuccess(applicationDeploymentListener, emptyAppFileBuilder.getId());
+        assertAppsDir(NONE, new String[] {emptyAppFileBuilder.getId()}, true);
```

Comment:
I suppose this checks that the test fails for the correct reason, and not a problem in the original application (if we could just test direct deployment this wouldn't be necessary). It was also in the other test.